autopush/endpoint.py

```diff
@@ -649,12 +649,11 @@ def post(self, router_type="", router_token="", uaid="", chid=""):
             return self._write_response(
                 400, 108, message="Invalid router")
         router = self.ap_settings.routers[router_type]
-        valid, router_token = router.check_token(router_token)
+        relo_url = self._relocate(router_type, uaid, chid)
+        valid, writer = router.check_token(router_token, self._write_response,
```

Comment:
I'm very nervous about inverting control of output writing to a router interface function. It drops the separation of responsibility for what chunk of code is responsible for which portion of the process.